File: list_products.js
import fs from 'fs';
import path from 'path';

const rootDir = "C:\\Users\\İsmail\\Desktop\\DORUK SÖVE KURUMSAL\\DORUK SÖVE ÜRÜNLER";

function listFiles(dir, level = 0) {
    const indent = '    '.repeat(level);
    const basename = path.basename(dir);
    console.log(`${indent}${basename}/`);

    if (!fs.existsSync(dir)) {
        console.log(`${indent}  (Not Found)`);
        return;
    }

    const items = fs.readdirSync(dir);
    items.forEach(item => {
        const fullPath = path.join(dir, item);
        const stats = fs.statSync(fullPath);
        if (stats.isDirectory()) {
            listFiles(fullPath, level + 1);
        } else {
            console.log(`${indent}    ${item}`);
        }
    });
}

listFiles(rootDir);

File: list_products.py
import os

root_dir = r"C:\Users\İsmail\Desktop\DORUK SÖVE KURUMSAL\DORUK SÖVE ÜRÜNLER"

def list_files(startpath):
    for root, dirs, files in os.walk(startpath):
        level = root.replace(startpath, '').count(os.sep)
        indent = ' ' * 4 * (level)
        print(f'{indent}{os.path.basename(root)}/')
        subindent = ' ' * 4 * (level + 1)
        for f in files:
            print(f'{subindent}{f}')

if os.path.exists(root_dir):
    list_files(root_dir)
else:
    print(f"Directory not found: {root_dir}")
